File: Strings/Two_Pointers/344_ReverseString.py

def reverseString(s):
    """
    Do not return anything, modify s in-place instead.
    """
    # Two pointer way:

    # left, right = 0, len(s) - 1
    # while left < right:
    #   s[left], s[right] = s[right], s[left]
    #   left += 1
    #   right -= 1

    # return s
    
    # Pythonic way....
    for i in range(len(s)//2):
        s[i], s[~i] = s[~i], s[i]
        
    return s


def main():
  print(reverseString(["h", "e", "l", "l", "o"]))
  print(reverseString(["k", "e", "v", "i", "n"]))


main()

File: Strings/Two_Pointers/344_ReverseString.js
const reverseString = (s) => {
  let left = 0,
    right = s.length - 1;

  while (left < right) {
    const temp = s[right];
    s[right] = s[left];
    s[left] = temp;
    left++;
    right--;
  }

  return s;
};
